framework/test/unit/modules/interoperability/mainchain/module.spec.ts

```diff
@@ -729,5 +763,244 @@ must NOT have more than ${MAX_NUM_VALIDATORS} items`,
 				await expect(interopMod.initGenesisState(context)).resolves.toBeUndefined();
 			});
 		});
+
+		describe('terminatedStateAccounts', () => {
+			it('should throw error if chainInfo.chainID exists in terminatedStateAccounts & chainInfo.chainData.status !== CHAIN_STATUS_TERMINATED', async () => {
```

Comment:
We should add one more test to throw error in case chainInfo.chainID does not exist at all and a terminated state account exists.. (recall in LIP it is ["if and only if"](https://github.com/LiskHQ/lips/blob/6796693a0aaf97c8a7dd63cbc73f923ca5a39f53/proposals/lip-0045.md?plain=1#L1563))

Reply:
In fact we should make sure that the implementation also covers this case. It seems to me that currently it does not (see comment in module.ts file).

Reply:
I think there is still one check missing (to completely check the "if and only if"): Check that for every entry in `chainInfo` in `chainInfos` such that `chainInfo.chainData.status`  is terminated, there exists a terminated state account for `chainInfo.chainID`. This is correctly enforced in the implementation in (module.ts, lines 465-473).

Reply:
To be completely clear the "if and only if" implies the following:

1) If there exists an entry `chainInfo` in `chainInfos` such that `chainInfo.chainData.status` is terminated, there exists a terminated state account for `chainInfo.chainID` (this check seems to be missing).
2) If there exists an entry `terminatedState` in `terminatedStateAccounts`, then there should be an entry in `chainInfos` with `chainID = terminatedState.chainID` and status terminated. So you need to check that an error is thrown if
    1) such an entry does not exist in `chainInfos` (you do it in lines 818-853) and
    2) such an entry exists but does not have status "terminated" ( you do it in lines 782-816)

Reply:
Now all checks seem to be in place and I am resolving this comment.